commands_test.go

```diff
@@ -293,3 +296,49 @@ func TestRunActionForeachMachine(t *testing.T) {
 		}
 	}
 }
+
+func TestCmdConfig(t *testing.T) {
+	stdout := os.Stdout
+
+	if err := clearHosts(); err != nil {
+		t.Fatal(err)
+	}
+
+	os.Setenv("MACHINE_STORAGE_PATH", TestStoreDir)
+
+	flags := getDefaultTestDriverFlags()
+
+	store := NewStore(TestMachineDir, "", "")
+	var err error
+
+	_, err = store.Create("test-a", "none", flags)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	host, err := store.Load("test-a")
+	if err != nil {
+		t.Fatalf("error loading host: %v", err)
+	}
+
+	if err := store.SetActive(host); err != nil {
+		t.Fatalf("error setting active host: %v", err)
+	}
+
+	set := flag.NewFlagSet("config", 0)
+
+	testOutput := &bytes.Buffer{}
+
+	go io.Copy(testOutput, stdout)
+
+	c := cli.NewContext(nil, set, set)
+
+	cmdConfig(c)
+
+	if strings.Contains(testOutput.String(), "-H=unix:///var/run/docker.sock") {
+		t.Fatalf("Expect docker host")
+	}
+
+	os.Stdout = stdout
+	os.Setenv("MACHINE_STORAGE_PATH", "")
```

Comment:
Should these type of "undo" things be in some `defer`s so that when the t.Fatal's are run we still reset the env?